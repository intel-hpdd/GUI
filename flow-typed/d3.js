// @flow

declare module d3 {
  declare module.exports: {
    max(xs:number[]):number;
    extent(xs:number[]):number[];
    scale:Object;
    time:Object;
  }
}
