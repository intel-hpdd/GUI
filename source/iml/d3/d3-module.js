// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import d3 from 'd3';

export default angular.module('d3', []).value('d3', d3).name;
