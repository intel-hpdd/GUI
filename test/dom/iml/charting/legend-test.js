// @flow

import Inferno from "inferno";

import Legend from "../../../../source/iml/charting/legend.js";
import d3 from "d3";

it("should render as expected", () => {
  const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");

  Inferno.render(
    <Legend
      svg={d3.select(svg).datum([0, 100])}
      dimensions={{ usableWidth: 200, usableHeight: 200 }}
      colors={d3.scale
        .ordinal()
        .domain(["foo", "bar"])
        .range(["green", "blue"])}
    />,
    svg
  );

  expect(svg).toMatchSnapshot();
});

it("should transform", () => {
  const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");

  Inferno.render(
    <Legend
      svg={d3.select(svg).datum([0, 100])}
      dimensions={{ usableWidth: 200, usableHeight: 200 }}
      transform="translate(0,100)"
      colors={d3.scale
        .ordinal()
        .domain(["foo", "bar"])
        .range(["green", "blue"])}
    />,
    svg
  );

  expect(svg).toMatchSnapshot();
});
