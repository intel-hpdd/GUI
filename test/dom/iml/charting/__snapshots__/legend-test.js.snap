// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<svg>
  <g
    class="legend chart-legend"
  >
    <g
      class="legend-group"
      transform="translate(0,0)"
    >
      <circle
        class="legend-circle"
        fill="green"
        fill-opacity="0"
        r="5"
        stroke="green"
      />
      <text
        class="legend-label"
        display="inherit"
        dx="10"
        dy="4"
      >
        foo
      </text>
    </g>
    <g
      class="legend-group"
      transform="translate(20,0)"
    >
      <circle
        class="legend-circle"
        fill="blue"
        fill-opacity="0"
        r="5"
        stroke="blue"
      />
      <text
        class="legend-label"
        display="inherit"
        dx="10"
        dy="4"
      >
        bar
      </text>
    </g>
  </g>
  
</svg>
`;

exports[`should transform 1`] = `
<svg>
  <g
    class="legend chart-legend"
    transform="translate(0,100)"
  >
    <g
      class="legend-group"
      transform="translate(0,0)"
    >
      <circle
        class="legend-circle"
        fill="green"
        fill-opacity="0"
        r="5"
        stroke="green"
      />
      <text
        class="legend-label"
        display="inherit"
        dx="10"
        dy="4"
      >
        foo
      </text>
    </g>
    <g
      class="legend-group"
      transform="translate(20,0)"
    >
      <circle
        class="legend-circle"
        fill="blue"
        fill-opacity="0"
        r="5"
        stroke="blue"
      />
      <text
        class="legend-label"
        display="inherit"
        dx="10"
        dy="4"
      >
        bar
      </text>
    </g>
  </g>
  
</svg>
`;
